cql-pytest: test a few small materialized views CQL issues

nyh opened the pull request:
While documenting materialized view in a new document (Refs #16569) I encountered a few questions on how various CQL operations work on a table that has views, and this patch contains tests that clarify their answer - and can later guarantee that the answer doesn't unintentionally change in the future. The questions that these tests answer are:

1. That TRUNCATE on a base table also TRUNCATEs its views. This is just a basic test, with no attempt to reproduce issue #17635 (which is about the truncation of the base and views not being atomic).

2. That DROP TABLE is *not allowed* on a base table that has views.

3. That DROP KEYSPACE is allowed, even if there are tables with views.

4. Test that ALTER TABLE tbl DROP is never allowed in Cassandra, but allowed in some cases by Scylla
    
5. Test that ALTER TABLE tbl ADD is allowed, and "SELECT *" expands to select the new column into the materialized view as well.

All the new tests pass on both Scylla and Cassandra.

mykaul commented on test/cql-pytest/test_materialized_view.py:
```diff
@@ -1354,3 +1354,60 @@ def test_create_materialized_view_slash_name(cql, test_keyspace, table1):
         cql.execute(f'DROP MATERIALIZED VIEW IF EXISTS {test_keyspace}."/xyz/"')
+
+# Test that TRUNCATE on a base table also truncates its views. This test
+# try to exercise the inconsistency created by non-atomic deletion of a
```
```suggestion
# tries to exercise the inconsistency created by non-atomic deletion of a
```

nyh replied:
Actually, doesn't try :-) I'll fix.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12465/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12465/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12465/)
#### Failed Tests (4/28131):
- [test_truncate_base.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/12465/testReport/junit/%28root%29/test_materialized_view/test_truncate_base_debug_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_truncate_base.debug.1)
- [test_truncate_base.debug.2](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/12465/testReport/junit/%28root%29/test_materialized_view/test_truncate_base_debug_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_truncate_base.debug.2)
- [cql-pytest.test_materialized_view.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/12465/testReport/junit/%28root%29/non-boost%20tests/cql_pytest_test_materialized_view_debug_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+cql-pytest.test_materialized_view.debug.1)
- [cql-pytest.test_materialized_view.debug.2](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/12465/testReport/junit/%28root%29/non-boost%20tests/cql_pytest_test_materialized_view_debug_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+cql-pytest.test_materialized_view.debug.2)
#### Build Details:
- Duration: 7 hr 14 min
- Builder: i-0f9d53c15ba631d82 (m5ad.8xlarge)

nyh commented:
Unfortunately, the new truncate test is flaky because it reproduces https://github.com/scylladb/scylladb/issues/17635:
* We create an empty view. View building starts, but takes it's time to even start processing the table...
* Meanwhile we write data into the table
* View building finally remembers to wake up, and finds data in the table and starts to build it, but in the meantime...
* The test runs TRUNCATE, which deletes the base and the view data
* The view building sends some of the data it was working on to the view.
* At this point the base table is empty, the view is not.

To unflake this test and make it not depend on https://github.com/scylladb/scylladb/issues/17635, I'll need to check that view building on the empty base completed before doing the TRUNCATE

nyh commented:
Rebased to fix conflicts and also fixed the flaky test that ran into https://github.com/scylladb/scylladb/issues/17635 by waiting for the view building to finish before testing TRUNCATE (as the commit message explained, the main idea of that test was to test that the TRUNCATE operation when #17635 didn't apply - running into it yet again was unplanned.

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12798/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12798/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12798/)
#### Build Details:
- Duration: 3 hr 52 min
- Builder: spider3.cloudius-systems.com
